#! /bin/sh -e

# DP: - When running the ACATS, look for the gnat tools in their new
# DP:   directory (build/gnattools), and for the shared libraries in
# DP:   build/gcc/ada/rts, build/libgnatvsn and build/libgnatprj.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- gcc/testsuite/ada/acats/run_acats~
+++ gcc/testsuite/ada/acats/run_acats	2006-04-22 08:57:50.933824704 +0200
@@ -11,23 +11,42 @@
 host_gnatmake=`type gnatmake | awk '{print $3}'`
 ROOT=`${PWDCMD-pwd}`
 BASE=`cd $ROOT/../../..; ${PWDCMD-pwd}`
-
-PATH=$BASE:$ROOT:$PATH
-ADA_INCLUDE_PATH=$BASE/ada/rts
+GNATTOOLS=`cd $BASE/../gnattools; ${PWDCMD-pwd}`
+LIBGNATVSN=`cd $BASE/../libgnatvsn; ${PWDCMD-pwd}`
+LIBGNATPRJ=`cd $BASE/../libgnatprj; ${PWDCMD-pwd}`
+
+PATH=$GNATTOOLS:$BASE:$ROOT:$PATH
+ADA_INCLUDE_DIR=$BASE/ada/rts
+ADA_INCLUDE_PATH=$ADA_INCLUDE_DIR:$LIBGNATVSN:$LIBGNATPRJ
 LD_LIBRARY_PATH=$ADA_INCLUDE_PATH:$BASE:$LD_LIBRARY_PATH
 ADA_OBJECTS_PATH=$ADA_INCLUDE_PATH
 
-if [ ! -d $ADA_INCLUDE_PATH ]; then
+if [ ! -d $ADA_INCLUDE_DIR ]; then
    echo gnatlib missing, exiting.
    exit 1
 fi
 
-if [ ! -f $BASE/gnatchop ]; then
+if [ ! -d $GNATTOOLS ]; then
+   echo gnattools missing, exiting.
+   exit 1
+fi
+
+if [ ! -d $LIBGNATVSN ]; then
+   echo libgnatvsn missing, exiting.
+   exit 1
+fi
+
+if [ ! -d $LIBGNATPRJ ]; then
+   echo libgnatprj missing, exiting.
+   exit 1
+fi
+
+if [ ! -f $GNATTOOLS/gnatchop ]; then
    echo gnattools missing, exiting.
    exit 1
 fi
 
-if [ ! -f $BASE/gnatmake ]; then
+if [ ! -f $GNATTOOLS/gnatmake ]; then
    echo gnattools missing, exiting.
    exit 1
 fi
